module/VuFindAdmin/src/VuFindAdmin/Controller/MaintenanceController.php

```diff
@@ -50,10 +50,62 @@ public function homeAction()
         $view = $this->createViewModel();
         $view->caches = $this->serviceLocator->get(\VuFind\Cache\Manager::class)
             ->getCacheList();
+        $view->scripts = $this->getScripts();
         $view->setTemplate('admin/maintenance/home');
         return $view;
     }
 
+    /**
+     * Get a list of the names of scripts available to run thorugh the admin panel.
+     *
+     * @return array
+     */
+    protected function getScripts(): array
+    {
+        // Load the AdminScripts.ini settings
+        $config = $this->serviceLocator->get(\VuFind\Config\PluginManager::class)
+            ->get('AdminScripts')->toArray();
+
+        // Filter out any commands that the current user does not have permission to run:
+        $permission = $this->permission();
+        $filter = function ($script) use ($permission) {
+            return !isset($script['permission'])
```

Comment:
I'd make permission mandatory for an added layer of security.

Reply:
My original thought was that you already need permission to access the admin module, so in some scenarios, that may be enough without being forced to add extra layers. However, it is relatively easy to just turn on "admin_enabled" in config.ini without setting proper permissions in permissions.ini, so I agree that we should probably be more strict. However, I see two possible approaches:

1.) If permission is undefined, default to "Access.AdminModuleScriptRunner" so there's a single permission to configure if granularity is not important (which makes the configuration more concise).

2.) Force a permission to be explicitly defined for every block in the configuration.

I somewhat prefer option 1, but option 2 is easy enough if you think it's better!

Reply:
Option 1 is fine!

Reply:
Though on a second thought, I think not requiring particular permission when you can already access the admin UI is quite fine as well.

Reply:
Thanks, @EreMaijala. I suppose we could also potentially add a "default_script_runner_permission" setting to config.ini (or something like that) to establish a configurable default permission... Or we could create a [General] or [Global] section of AdminScripts.ini for global configuration settings like that. Do any of those ideas particularly appeal (or not appeal) to you?

Reply:
How does [ca260f0](https://github.com/vufind-org/vufind/pull/2866/commits/ca260f0df8b012f9c77e59167d5c50695035956a) look to you?